### ISSUE TITLE
Working with Google Cloud TPUs

### ISSUE THREAD

**Issue by sercanamac** [allenai/allennlp#2801]
**System (please complete the following information):**
Google Cloud

**Question**
Hi can we make allennlp work on Cloud TPUs? Thanks in advance.

**Comment by DeNeutoy** [comment]
No, not in the near future. Pytorch for TPU is in very early alpha.

**Comment by StrangeTcy** [comment]
How about now? 
While not as easy as writing `model.to(torch.device('tpu'))`, it's definitely not in alpha anymore. There are computer vision models one could train and run easily on colab/kaggle TPUs, as well as transformer models.

**Comment by epwalsh** [comment]
This is definitely worth revisiting.

**Comment by vikigenius** [comment]
@epwalsh any further progress on this. It seems like to be able to use TPUs we need to use xla devices from the torch_xla package. https://pytorch.org/xla/release/1.9/index.html. 

This would be a great feature to have.

**Comment by epwalsh** [comment]
@vikigenius no updates yet, but that's great news that the PyTorch XLA API seems to be stabilizing.
